share/namespace.go

```diff
@@ -92,35 +116,60 @@ func (n Namespace) ID() []byte {
 	return n.data[NamespaceVersionSize:]
 }
 
+// String stringifies the Namespace.
+func (n Namespace) String() string {
+	return hex.EncodeToString(n.data)
+}
+
 // ValidateUserNamespace returns an error if the provided version is not
 // supported or the provided id does not meet the requirements
 // for the provided version. This should be used for validating
 // user specified namespaces
-func ValidateUserNamespace(version uint8, id []byte) error {
-	err := validateVersionSupported(version)
+func (n Namespace) ValidateUserNamespace() error {
```

Comment:
nit: can we just call this `Validate`. `ValidateUserNamespace` seems misleading because this also applies to namespaces that users can't use (like reserved namespaces or tailpadding)

Reply:
renamed and made private